### ISSUE TITLE
Payment not working

### ISSUE THREAD

**Issue by muruganupturn** [princeflutterdev/flutter_upi_pay#4]
Your money has not been debited. You've exceeded thebank limit for this payment. Retry with a smaller amount

**Comment by muruganupturn** [comment]
Now come to Recipient 's bank is currently unable to receive the money. But normally  GPay is send money

**Comment by muruganupturn** [comment]
The CRED app using payment is success. But transaction status error 
![Screenshot 2024-10-10 at 5 46 46 PM](https://github.com/user-attachments/assets/f9c80bf1-105c-48b5-8958-5c85b2d5d956)

**Comment by princeflutterdev** [comment]
Can you able to provide me the CRED response what you getting.

**Comment by Hit1009** [comment]
same issue, showing error even on merchant id. Tried both on personal UPI and merchant UPI I'd. Is it working for you @princeflutterdev?

**Comment by Hit1009** [comment]
Also is there a way to see the final upi link generated for payment in the format of upi://pay?pa=xyz@paytm&pn=Paytm%20Merchant&mc=abc&mode=02&orgid=000000&paytmqr=abcdefg&sign=abc+xyz/pqr/+stu? Most possibly issue in the link that redirects us to the upi app.

**Comment by pavan-jaju** [comment]
Any update on this?

**Comment by princeflutterdev** [comment]
> Also is there a way to see the final upi link generated for payment in the format of `upi://pay?pa=xyz@paytm&pn=Paytm%20Merchant&mc=abc&mode=02&orgid=000000&paytmqr=abcdefg&sign=abc+xyz/pqr/+stu` Most possibly issue in the link that redirects us to the upi app.

yes there is a way but you can download the code of this package and run it in your system and do changes

**Comment by RupakFlutterDeveloper** [comment]
import 'package:flutter/material.dart';
import 'package:flutter_upi_india/flutter_upi_india.dart';

class UpiScreen extends StatelessWidget {
  // Fetch the available UPI apps and print the list
  void _openUpiApp() async {
    try {
      // Get the installed UPI apps using UpiPay
      List<ApplicationMeta> upiApps =
          await UpiPay.getInstalledUpiApplications();

      if (upiApps.isNotEmpty) {
        for (var app in upiApps) {
          print('UPI App Name: ${app.packageName}');
          print('Package Name: ${app.packageName}');
        }
      } else {
        print('No UPI apps installed');
      }
    } catch (e) {
      // Catch and print any errors
      print('Error while fetching UPI apps: $e');
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Open UPI App"),
        centerTitle: true,
      ),
      body: Center(
        child: ElevatedButton(
          onPressed: _openUpiApp,
          child: Text("Check UPI Apps"),
        ),
      ),
    );
  }
}
but still issue I/flutter (18770): No UPI apps installed  what to do

**Comment by dhruvpatidar359** [comment]
Hi, here phonepe says you have only 2000rs limit . Similarly others also give some errors and thus it does not works

**Comment by Hit1009** [comment]
Hello, I think the issue is that UPI apps/banks has blocked direct link payments from unsolicited sources. Only QR code and links from trusted payment gateways( like razorpay, paytm etc) are allowed. A payment will get executed only if you scan the QR using built-in UPI app scanner. Using direct links of the same QR code or google lens will lead to these errors.

**Comment by SuTechs** [comment]
For now it seems there's no solution. One workaround is to pass merchant upi id but even that doesn't work on many devices and upi apps.

**Comment by princeflutterdev** [comment]
> For now it seems there's no solution. One workaround is to pass merchant upi id but even that doesn't work on many devices and upi apps.

yes this will work only for QRv2 only .. not merchant qr or UPI will not work . QRv2 like RazorPay Gateway , Paytm Payment Gateway, HDFC v2 Merchant will work 

if you want to know how that will work dm me i will share the details with you
